src/validation/tests/statement_validation_tests.rs

```diff
@@ -1468,3 +1468,57 @@ fn for_loop_conditions_are_real_and_trigger_error() {
 
     "###);
 }
+
+#[test]
+fn if_statement_triggers_error_if_condition_is_not_boolean() {
+    let diagnostic = parse_and_validate_buffered(
+        "
+        FUNCTION main
+            VAR
+                x : BOOL;
+                y : DINT;
+                z : STRING;
+            END_VAR
+
+            IF      x THEN
+            ELSIF   y THEN
+            ELSIF   z THEN
+            ELSIF   0 THEN
+            ELSIF   1 THEN
+
+            // These should be Ok
+            ELSIF   (0 < 1)         THEN
+            ELSIF   (y < 0)         THEN
+            ELSIF   ((1 = 2) = 3)   THEN
+            END_IF
+        END_FUNCTION
+        ",
+    );
+
+    assert_snapshot!(diagnostic, @r###"
+    error: Expected a boolean, got `DINT`
+       ┌─ <internal>:10:21
+       │
+    10 │             ELSIF   y THEN
+       │                     ^ Expected a boolean, got `DINT`
+
+    error: Expected a boolean, got `STRING`
+       ┌─ <internal>:11:21
+       │
+    11 │             ELSIF   z THEN
+       │                     ^ Expected a boolean, got `STRING`
+
+    error: Expected a boolean, got `DINT`
+       ┌─ <internal>:12:21
+       │
+    12 │             ELSIF   0 THEN
```

Comment:
I think IF 1 and IF 0 are fine.. not because i like them but because 0 and 1 are common I think

Reply:
I initially also wanted to support them but I thought if someone is using 0 or 1 they could also just use FALSE or TRUE instead. But also I just replicated what Rust does, though comparing with C makes more sense generally speaking. Anyhow, do we proceed with supporting 0 and 1s here?